modules/realvuAnalyticsAdapter.js

```diff
@@ -0,0 +1,849 @@
+    }
+    return 'nr';
+  },
+
+  getStatusById: function(unit_id) { // Jun 1, 2015 - return status object
+    for (var i = 0; i < this.ads.length; i++) {
+      var adi = this.ads[i];
+      if (adi.unit_id == unit_id) return this.fmt(adi);
+    }
+    return null;
+  },
+
+  log: function(m1, i) {
+    if (this.doLog) {
+      this.msg.push({
+        dt: new Date() - this.t0,
+        s: 'U' + (i + 1) + m1
+      });
+    }
+  },
+
+  keyPos: function(a) {
+    if (a.pins[0].unit_id) {
+      var level = 'L' + (window.top1.location.pathname.match(/\//g) || []).length;
+      return 'realvu.' + level + '.' + a.pins[0].unit_id.replace(/[0-9]{5,}/gi, 'RANDOM');
+    }
+  },
+  writePos: function(a) {
+    var v = a.x + ',' + a.y + ',' + a.w + ',' + a.h;
+    if (localStorage) localStorage.setItem(this.keyPos(a), v);
+  },
+  readPos: function(a) {
+    if (localStorage) {
+      var s = localStorage.getItem(this.keyPos(a));
+      if (s) {
+        var v = s.split(',');
+        a.x = parseInt(v[0], 10);
+        a.y = parseInt(v[1], 10);
+        a.w = parseInt(v[2], 10);
+        a.h = parseInt(v[3], 10);
+        a.box = {x: a.x, y: a.y, w: a.w, h: a.h};
+        return true;
+      }
+    }
+    return false;
+  },
+
+  keyMem: function(num) {
+    return this.keyPos(this.ads[num]) + '.score';
+  },
+
+  incrMem: function(num, v) {
+    var k1 = this.keyMem(num);
+    var n = parseInt(this.getMem(k1), 10);
+    if (isNaN(n)) {
+      n = 0x57F57; // initial score 75, 20 bits, step 5
+    }
+    if (v == 'r')n = n << 1;
+    if (v == 'v')n |= 1;
+    n &= 0xFFFFF;
+    this.updateMem(k1, n);
+  },
+
+  score: function(num) {
+    var k1 = this.keyMem(num);
+    var r = parseInt(this.getMem(k1));
+    if (isNaN(r))r = 0x57F57;
+    var s = 0;
+    for (r &= 0x3FF; r > 0; r >>>= 1) {
+      if (r & 0x1)s++;
+    }
+    return s * 100.0 / 10;
+  },
+  getMem: function(key) {
+    return localStorage.getItem(key);
+  },
+  updateMem: function(key, value) {
+    localStorage.setItem(key, value);
+  }
+};
+
+if (typeof (window.top1.boost_poll) == 'undefined') {
+  window.top1.realvu_boost.init();
+  window.top1.boost_poll = setInterval(function() {
+    window.top1 && window.top1.realvu_boost && window.top1.realvu_boost.poll();
+  }, 20);
+}
+
+var options = { };
+
+realvuAnalyticsAdapter.originEnableAnalytics = realvuAnalyticsAdapter.enableAnalytics;
+
+realvuAnalyticsAdapter.enableAnalytics = function (config) {
+  options = config.options;
+  realvuAnalyticsAdapter.originEnableAnalytics(config);
+};
+
+realvuAnalyticsAdapter.track = function ({eventType, args}) {
+  var msg = document.getElementById('msg_an');
+  if (msg) {
+    msg.innerHTML += 'track: eventType=' + eventType + ', args=' + JSON.stringify(args) + '<br>';
+  }
+  if (eventType === CONSTANTS.EVENTS.AUCTION_INIT) {
+    if (options && options.partnerId) {
+      var hb = $$PREBID_GLOBAL$$;
```

Comment:
@tchibirev This is not allowed. For each event we are sending `args` you can only work with those.

Reply:
Hi jaiminpanchal27,  Is this constraint documented